classes/Tools.php

```diff
@@ -2218,6 +2218,10 @@ public static function generateHtaccess($path = null, $rewrite_settings = null,
         fwrite($write_fd, "# .htaccess automaticaly generated by PrestaShop e-commerce open-source solution\n");
         fwrite($write_fd, "# http://www.prestashop.com - http://www.prestashop.com/forums\n\n");
 
+        // prevent click-jacking
+        fwrite($write_fd, "Header append Content-Security-Policy \"frame-ancestors 'self';\"\n");
+        fwrite($write_fd, "Header append X-Frame-Options SAMEORIGIN\n\n");
```

Comment:
This should be surrounded by `<IfModule mod_headers.c>`, which already exist in this .htaccess.

Please move these lines further in the method, as all environment does not have the "header" module enabled (for instance, the docker images).